### PR TITLE
[SDK] Add init method in DefaultService

### PR DESCRIPTION
**Pull request by mbdcos**
DefaultService is initialized and started by init method. Framework service classes extending DefaultService will call this init method in their constructor.

### CODE REVIEW

**Comment by mbdcos** [comment]
retest this please

**Comment by mbdcos** [comment]
retest this please

**Comment by mbdcos** [comment]
retest this please

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultService.java [hunk 1]
```diff
@@ -48,6 +48,10 @@
 
     private ServiceSpec serviceSpec;
 
+    public DefaultService() throws Exception {
```
Comment (mohitsoni) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultService.java]: @mbdcos Why did you add a throws Exception to this method's signature? The body is empty. 

Can you please provide an example usage of this ?

Comment (mbdcos) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultService.java]: No need to throw exception.

**Comment by mbdcos** [comment]
retest this please

**Comment by mbdcos** [comment]
See PR #639
